sdcm/nemesis.py

```diff
@@ -834,14 +841,21 @@ def disrupt_snapshot_operations(self):
 
 
 class NotSpotNemesis(Nemesis):
-    def set_target_node(self):
-        if isinstance(self.cluster, ScyllaAWSCluster):
-            non_seed_nodes = [node for node in self.cluster.nodes if not node.is_seed and not node.is_spot]
-        else:
-            non_seed_nodes = [node for node in self.cluster.nodes if not node.is_seed]
-        if non_seed_nodes:
-            self.target_node = random.choice(non_seed_nodes)
+    def set_target_node(self, nemesis_start=False):
+
+        with self.lock:
+            if isinstance(self.cluster, ScyllaAWSCluster):
```

Comment:
It would be good if we add here comment that explains the logic and the limitations.
E.g. we assume here that we don't run two nemesis on the same node. 
It's ok to assume it for now, but we should be aware of it and mention it.
I can think of ways in the future we would want to improve it.
e.g. testing a "disruptive" nemesis on the same node that runs "non-disruptive" one and cause it to fail.
and to be more specific, running restart during cleanup.

Reply:
added comment